underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java

```diff
@@ -91,8 +91,10 @@ public boolean delete(String path, boolean recursive) throws IOException {
     boolean success = true;
     if (recursive && file.isDirectory()) {
       String[] files = file.list();
-      for (String child : files) {
-        success = success && delete(PathUtils.concatPath(path, child), true);
+      if (files != null) {
```

Comment:
Thanks for the detailed explanation, that makes sense now. Could you add a comment so that nobody accidentally removes the null check?